tests/unit/lms/resources/_js_config_test.py

```diff
@@ -27,6 +28,53 @@ def test_enable_content_item_selection_mode(self, js_config):
         assert js_config.config["mode"] == "content-item-selection"
 
 
+class TestMaybeSetFocusedUser:
+    def test_it_does_nothing_if_theres_no_focused_user_param(
+        self, js_config, pyramid_request
+    ):
+        del pyramid_request.params["focused_user"]
+
+        js_config.maybe_set_focused_user()
+
+        assert "focus" not in js_config.config["hypothesisClient"]
+
+    def test_it_sets_the_focused_username(self, js_config):
+        js_config.maybe_set_focused_user()
+
+        assert (
+            js_config.config["hypothesisClient"]["focus"]["user"]["username"]
+            == "example_h_username"
+        )
+
+    def test_it_gets_the_display_name_from_the_h_API(self, h_api, js_config):
+        js_config.maybe_set_focused_user()
+
+        h_api.get_user.assert_called_once_with("example_h_username")
+
+    def test_it_sets_the_focused_display_name(self, js_config):
+        js_config.maybe_set_focused_user()
+
+        assert (
+            js_config.config["hypothesisClient"]["focus"]["user"]["displayName"]
+            == "example_h_display_name"
+        )
+
+    def test_display_name_falls_back_to_a_default_value(self, h_api, js_config):
+        h_api.get_user.side_effect = HAPIError()
+
+        js_config.maybe_set_focused_user()
+
+        assert (
+            js_config.config["hypothesisClient"]["focus"]["user"]["displayName"]
+            == "(Couldn't fetch student name)"
+        )
+
+    @pytest.fixture
+    def pyramid_request(self, pyramid_request):
+        pyramid_request.params["focused_user"] = "example_h_username"
+        return pyramid_request
```

Comment:
This very micro unittests are great for highlighting precise issues, and they are very easy to read, but they do get _affa_ long. 

I think we should be alive to different ways of expressing the same thing with fewer cases, or in a more compact way. For example, I think a "it_stores_user_details_correctly" that covered both user and display name would collapse two of these into a pretty simple test case.

Reply:
Collapsed these into one test with multiple asserts

Reply:
I don't think there's any loss of clarity here, probably the reverse as you can see the whole data structure which is set in one go.